### DISCUSSION TITLE
Mini-Cart console warnings, missing X-mark icon (mobile), and overflow issues

### DISCUSSION THREAD

**Discussion by daotw** [woocommerce/woocommerce#45101] [WooCommerce, Blocks, and Site Editing]
Below screenshots are recorded without any plugin activated (except Woo) and with twenty twenty-four theme.

Desktop view: when loading the page ↓
<img width="1410" alt="console-1" src="https://github.com/woocommerce/woocommerce/assets/31888729/501ceeb7-c1aa-42e0-869e-df256ca7255f">
"The resource <URL> was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally."

Desktop view: when hovering over mini-cart icon ↓
<img width="1425" alt="console-2" src="https://github.com/woocommerce/woocommerce/assets/31888729/0467d276-d930-4f9b-a98e-d4cb8c94718b">
"Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17."

Mobile view: even more warnings when clicking/tapping mini-cart icon ↓
<img width="1423" alt="console-3" src="https://github.com/woocommerce/woocommerce/assets/31888729/048ba5a7-2f1b-4b10-91ec-b64a4611870f">

From a visitor's perspective, Mini-Cart block works fine on desktop, but buggy (no X-mark icon for closing) on mobile.

I can't find related issues so I'm opening one.

Thank you.

**Comment by daotw** [comment]
Mini-Cart also causes `<body>` overflow when scrolling down at frontend. Console warnings, missing X-mark icon (mobile), and overflow issues are gone when Mini-Cart block is removed. Hope there will be fixes 🙏

**Comment by ralucaStan** [comment]
@daotw thank you for taking the time to surface your findings.

> I can't find related issues so I'm opening one.
For opening issues we recommend using the [New Issue page ](https://github.com/woocommerce/woocommerce/issues/new/choose). 

I took a look at the Mini Cart using the WooCommerce 8.8 RC. 

> Desktop view: when loading the page ↓

I was able to replicate this. I opened an issue for this https://github.com/woocommerce/woocommerce/issues/46332

> Desktop view: when hovering over mini-cart icon ↓

I could not replicate this.

> Mobile view: even more warnings when clicking/tapping mini-cart icon ↓

I was not able to replicate this.

> From a visitor's perspective, Mini-Cart block works fine on desktop, but buggy (no X-mark icon for closing) on mobile.

The Mini Cart has a close functionality when opened on Mobile
![Screenshot 2024-04-08 at 11 21 20](https://github.com/woocommerce/woocommerce/assets/1628454/847cd167-11b5-4d6e-9d15-acaeaa28ab79)

Considering all the above I am going to close this discussion.

**Reply by daotw** [reply]
@ralucaStan Thank you for opening an issue!

When mini-cart is present in `<header>`, this CSS causes horizontal overflow:

```css
header {
    transform: translateY(-100%);
}
```

I just replicated the issue with **fresh WP + fresh 2024 theme + fresh Woo**.

When mini-cart is not present in `<header>`, this CSS doesn't cause horizontal overflow, of course.

When mini-cart-caused horizontal overflow is present, mouse hovering over mini-cart will cancel horizontal overflow.

Devs who want to implement sticky-header-like feature will certainly encounter this issue.

**Reply by ralucaStan** [reply]
Thank you @daotw for reaching out again. I saw you added this as a comment on an existing issue. https://github.com/woocommerce/woocommerce/issues/46332#issuecomment-2102974917 

Would you mind creating a separate issue https://github.com/woocommerce/woocommerce/issues/new?assignees=&labels=&projects=&template=1-bug-report.yml? That would be really helpful for us.
